File: school-management-backend/src/main/java/com/school/management/service/FeeService.java
package com.school.management.service;

import com.school.management.dto.FeeRequest;
import com.school.management.entity.Fee;

import java.util.List;

public interface FeeService {
    List<Fee> getAllFees();
    Fee getFeeById(Long id);
    Fee createFee(FeeRequest request);
    Fee updateFee(Long id, FeeRequest request);
    void deleteFee(Long id);
    List<Fee> getFeesByStudentId(Long studentId);
    List<Fee> getPendingFees();
}

File: school-management-frontend/src/services/feeService.js
import api from '../api/axios';

const feeService = {
  getAllFees: () => api.get('/fees'),
  getFeeById: (id) => api.get(`/fees/${id}`),
  getPendingFees: () => api.get('/fees/pending'),
  getFeesByStudent: (studentId) => api.get(`/fees/student/${studentId}`),
  createFee: (feeData) => api.post('/fees', feeData),
  updateFee: (id, feeData) => api.put(`/fees/${id}`, feeData),
  deleteFee: (id) => api.delete(`/fees/${id}`),
};

export default feeService;
